### PR TITLE
py-flatten-dict: new recipe

### PR DESCRIPTION
**Pull request by Andrew-Dunning-NNL**
Add new python packaged for flatten-dict

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    in Python"""
+
+    homepage = "https://github.com/ianlini/flatten-dict"
+    pypi     = "flatten-dict/flatten-dict-0.3.0.tar.gz"
+
+    version('0.3.0', sha256='0ccc43f15c7c84c5ef387ad19254f6769a32d170313a1bcbf4ce582089313d7e')
+
+    depends_on('python@3.5:', type=('build', 'run'))
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-six', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-flatten-dict/package.py]:
```suggestion
    depends_on('py-six@1.12:1.999', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on('py-pathlib2', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-flatten-dict/package.py]:
```suggestion
    depends_on('py-pathlib2@2.3:2.999', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class PyFlattenDict(PythonPackage):
+    """A flexible utility for flattening and unflattening dict-lik objects
+    in Python"""
+
+    homepage = "https://github.com/ianlini/flatten-dict"
+    pypi     = "flatten-dict/flatten-dict-0.3.0.tar.gz"
+
+    version('0.3.0', sha256='0ccc43f15c7c84c5ef387ad19254f6769a32d170313a1bcbf4ce582089313d7e')
+
+    depends_on('python@3.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-flatten-dict/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-flatten-dict/package.py]: Are you sure this is needed at run-time?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py (hunk 4 above)
Comment (Andrew-Dunning-NNL) [reply var/spack/repos/builtin/packages/py-flatten-dict/package.py]: Removed runtime dependency

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py (hunk 3 above)
Comment (Andrew-Dunning-NNL) [reply var/spack/repos/builtin/packages/py-flatten-dict/package.py]: Added python 2 dependency

File: var/spack/repos/builtin/packages/py-flatten-dict/package.py (hunk 2 above)
Comment (Andrew-Dunning-NNL) [reply var/spack/repos/builtin/packages/py-flatten-dict/package.py]: added Versions

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**